app/javascript/lib/service-worker-cache.js

```diff
@@ -0,0 +1,55 @@
+import { getSCPContext } from '~/providers/SCPContextProvider'
+
+const scpContext = getSCPContext()
+const env = scpContext.environment
+const version = scpContext.version
+
+/** Whether to fetch data from service worker cache */
+export const isServiceWorkerCacheEnabled = scpContext.isServiceWorkerCacheEnabled
+
+// Cache keys are like a mix of database server and collection/table names:
+// they identify a cache store for individual cache entries.  Here we set up
+// cache keys specific to each SCP environment and version.
+const serviceWorkerCacheKeyStem = `scp-${env}`
+const serviceWorkerCacheKey = `${serviceWorkerCacheKeyStem}-${version}`
+
+/**
+ * Fetch, leveraging service worker cache if enabled and available
+ *
+ * TODO (SCP-4508): Account for same URL, different sign-in state in service worker cache
+ */
+export async function fetchServiceWorkerCache(url, init) {
+  const swCache = await caches.open(serviceWorkerCacheKey)
+  let response = await swCache.match(url)
+  let hitOrMiss = 'hit'
```

Comment:
BLOCKER: we shouldn't use magic strings to track boolean state
```suggestion
  let isHit = true
```

Reply:
Good catch, this was a vestige from early work.  Refactored in 0c53771.